Source/WebKit/GPUProcess/graphics/ImageBufferShareableAllocator.cpp

```diff
@@ -32,20 +32,43 @@
 #include <WebCore/ImageBuffer.h>
 #include <wtf/TZoneMallocInlines.h>
 
+#if HAVE(IOSURFACE)
+#include <WebCore/ImageBufferIOSurfaceBackend.h>
+#endif
+
 #if ENABLE(GPU_PROCESS)
 
 namespace WebKit {
 using namespace WebCore;
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(ImageBufferShareableAllocator);
 
+#if HAVE(IOSURFACE)
+ImageBufferShareableAllocator::ImageBufferShareableAllocator(const ProcessIdentity& resourceOwner, IOSurfacePool* ioSurfacePool)
+    : m_resourceOwner(resourceOwner)
+    , m_ioSurfacePool(ioSurfacePool)
+{
+}
+#else
 ImageBufferShareableAllocator::ImageBufferShareableAllocator(const ProcessIdentity& resourceOwner)
     : m_resourceOwner(resourceOwner)
 {
 }
+#endif
 
-RefPtr<ImageBuffer> ImageBufferShareableAllocator::createImageBuffer(const FloatSize& size, const DestinationColorSpace& colorSpace, RenderingMode) const
+RefPtr<ImageBuffer> ImageBufferShareableAllocator::createImageBuffer(const FloatSize& size, const DestinationColorSpace& colorSpace, RenderingMode renderingMode) const
 {
+#if HAVE(IOSURFACE)
+    if (renderingMode == RenderingMode::Accelerated) {
+        ImageBufferCreationContext creationContext;
+        creationContext.resourceOwner = m_resourceOwner;
+        if (m_ioSurfacePool)
+            creationContext.surfacePool = m_ioSurfacePool.get();
+
+        return ImageBuffer::create<ImageBufferIOSurfaceBackend>(size, 1, colorSpace, { PixelFormat::BGRA8 }, RenderingPurpose::Unspecified, creationContext);
+    }
```

Comment:
I haven't looked carefully but if it's possible to distinguish that this image buffer is specifically for filters it would be helpful to have a corresponding `RenderingPurpose` and `IOSurface::Name` so that we can distinguish the IOSurface's purpose in tools like `IOSDebug` or `vmmap`.

Reply:
i was wondering about that. fwiw, this sharable allocator seems to be used only for filtering purposes unless i'm missing a caller somewhere

Reply:
These buffers are short-lived, so I don't think tagging them is that important.